pkg/cmdlets/auth/login.go

```diff
@@ -178,3 +182,37 @@ func promptToken(credentials *mono_models.Credentials, out output.Outputer, prom
 
 	return nil
 }
+
+// AuthenticateWithDevice attempts to authenticate this device with the Platform.
+func AuthenticateWithDevice(out output.Outputer) error {
+	response, err := model.RequestDeviceAuthorization()
+	if err != nil {
+		return locale.WrapError(err, "err_auth_device")
+	}
+
+	if response.UserCode == nil || response.VerificationURIComplete == nil {
+		return locale.NewError("err_auth_device")
+	}
+
+	out.Notice(locale.Tr("auth_device_verify_security_code", *response.UserCode))
+
+	err = OpenURI(*response.VerificationURIComplete)
+	if err != nil {
+		logging.Warning("Could not open browser: %v", err)
+		out.Notice(locale.Tr("err_browser_open", *response.VerificationURIComplete))
+	}
+
+	authorization, err := model.WaitForAuthorization(response)
+	if err != nil {
+		return locale.WrapError(err, "err_auth_device")
+	}
+
+	err = authentication.LegacyGet().AuthenticateWithDevice(authorization.AccessToken)
```

Comment:
Still using legacy here.

Reply:
Right. This is where I was saying in an earlier comment that the code doesn't know whether it wants a `config.Instance` or a `keypair.Configurable`. With the help of your previous `primer.Author` comment I was able to fix this.